developer_manual/digging_deeper/flow.rst

```diff
@@ -2,8 +2,87 @@
+To integrate with the workflow engine, your application needs to implement event listeners and operations, and provide a configuration component for users. Event listeners are registered with the manager and listen for events like ``RegisterOperationsEvent``, where they register available operations and the name of the JavaScript bundle containing a Vue component for configuring conditions. Operations are classes that implement specific interfaces and define what happens when an event is triggered. The configuration component, built as a Vue app and registered via the JS bundle, provides the user interface for setting up and customizing flows. For more details, see the example workflow at https://github.com/nextcloud/workflow_pdf_converter, or watch the `36c3 talk <https://mirror.eu.oneandone.net/projects/media.ccc.de/congress/2019/h264-sd/36c3-oio-174-eng-Building_Nextcloud_Flow_sd.mp4>`_ and review the `slides <https://nextcloud.com/wp-content/themes/next/assets/files/Building_nextcloud_flow.pdf>`_.
 
-Contributions to this documentation are, as always, very welcome!
+Events
+======
+
+First, consider the events you want users to configure in your flow. Nextcloud currently dispatches the following events to the workflow engine:
+
+Files
+-----
+
+- ``\OCP\Files::postCreate``
+- ``\OCP\Files::postWrite``
+- ``\OCP\Files::postRename``
+- ``\OCP\Files::postDelete``
+- ``\OCP\Files::postTouch``
+- ``\OCP\Files::postCopy``
+
+Listeners
+=========
+
+Event listeners are classes registered with the manager, along with an array of events they should listen for.
+
+The first event listener your application should implement is for the ``RegisterOperationsEvent`` event.
+
+This event is dispatched when the workflow engine resolves which event listeners should receive a broadcasted event.
+
+This initial event listener should register your operations and the name of the JS bundle containing a Vue component for configuring conditions for your event listeners.
+
+`An example RegisterOperationsEvent listener can be found here <https://github.com/nextcloud/workflow_pdf_converter/blob/master/lib/Listener/RegisterFlowOperationsListener.php>`_.
+
+Register this event listener within the ``register`` function of your ``Application`` class::
+
+    public function register(IRegistrationContext $context): void {
+        $context->registerEventListener(
+            RegisterOperationsEvent::class, 
+            RegisterFlowOperationsListener::class
+        );
+    }
+
+Operations
+==========
+
+Operations are classes registered with one or more event listeners.
+
+This class must implement one of the following `IOperation <https://github.com/nextcloud/server/blob/master/lib/public/WorkflowEngine/IOperation.php#L33>`_ interfaces:
```

Comment:
They don't have to. Talk is implementing `IOperation` directly. Works as well. Only when you want to have the other things you need to implement those.

Reply:
Also the line looks unrelated here?